src/composite/MegaChart.js

```diff
@@ -103,7 +106,24 @@
             chartTypeSelect.change = function(a){
                 context.chartType(a.value()).render();
             };
-            this.toolbarWidgets([chartTypeSelect]);
+            
+            chartTypeSelect.change = function(a){
+                context.chartType(a.value()).render();
+            };
+            
+            var csvButton = new Button().value("CSV");
+            csvButton.click = function(a){
+                require(["src/common/Utility"],function(Util){
+                    Util.downloadData("CSV",[context._chart.columns()].concat(context._chart.data()).join("\r\n"));
+                });
+            };
+            if(this.showCSV()){
+                twArr.push(csvButton);
+            }
+            if(this.showChartSelect()){
+                twArr.push(chartTypeSelect);
+            }
+            this.toolbarWidgets(twArr);
             this.setContent("top", this._toolbar);
```

Comment:
This whole section really needs to be in the "update" method so that users can show/hide them in the dermatology editor.